lib/internal/webstreams/adapters.js

```diff
@@ -82,6 +86,28 @@ const { UV_EOF } = internalBinding('uv');
 
 const encoder = new TextEncoder();
 
+const ZLIB_FAILURES = new SafeSet([
+  ...ObjectEntries(internalBinding('constants').zlib)
+    .map(({ 0: code, 1: value }) => (value < 0 ? code : null)).filter(Boolean),
+  'Z_NEED_DICT',
+]);
+
+function handleKnownInternalErrors(cause) {
```

Comment:
Can you add jsdoc? Even though we don't enforce it, it would be nice to read the nullable case of this param.